contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -190,6 +191,13 @@ varbinaryin(PG_FUNCTION_ARGS)
 		PG_RETURN_BYTEA_P(result);
 	}
 
+	if (!(IsTdsClientHook && (*IsTdsClientHook)()) && 
```

Comment:
What you did before - using a rendevu variable - was better than adding a hook between 2 extensions.
And we are doing the same mistake here that we are gonna use byteain for not IsTdsClientHook == NULL, why would we want to keep that behvaiour in varbinaryin? We should not be intrusive with our changes, you are technically making varbinary unusable for people wanting to use only common extension. 
With your change anyone having only common extension and using binary will now be using byteain which is absolutely not what it is intended for

Reply:
The entire point is to use byteain when connecting through the PG endpoint - I just tested this, and even with all extensions installed, when connecting through the PG endpoint `PLtsql_protocol_plugin **protocol_ptr` is NULL. So there is no way to determine whether someone is using only the common extension or not. As mentioned before, there is a workaround if the user manually converts the input to hex first and inserts that as a BYTEA directly.